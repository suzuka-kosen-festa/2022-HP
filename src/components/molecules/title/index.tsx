import type { FC } from "react";
import tw, { css } from "twin.macro";
import { useMemo } from "react";
import type { TitleProperties } from "./types/model";

const TitleWrapper = tw.div`relative`;

const TitleContainer = tw.h1`absolute font-zen text-text font-bold select-none text-ink top-ink-top left-ink-left md:(text-ink-md top-ink-top-md left-ink-left-md)`;

const Title: FC<TitleProperties> = ({ children, colorKey, ...rest }) => {
  // eslint-disable-next-line consistent-return
  const inkPath = useMemo(() => {
    switch (colorKey) {
      case "yellow":
        return "/statics/Ink3_yellow.webp";
      case "green":
        return "/statics/Ink3_green.webp";
      case "blue":
        return "/statics/Ink3_blue.webp";
      case "purple":
        return "/statics/Ink3_purple.webp";
      case "red":
        return "/statics/Ink3_red.webp";
      // no default
    }
  }, [colorKey]);
  return (
    <TitleWrapper {...rest}>
      <img
        css={css`
          aspect-ratio: auto;
          width: 25%;
          @media (max-width: 768px) {
            width: 45%;
          }
        `}
        src={inkPath}
        alt="ink"
        width={592}
        height={580}
        loading="eager"
      />
      <TitleContainer>{children}</TitleContainer>
    </TitleWrapper>
  );
};

export { Title };
