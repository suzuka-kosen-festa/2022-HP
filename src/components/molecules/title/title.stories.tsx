import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { Title } from "./index";

type T = typeof Title;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  children: "This is test",
  colorKey: "yellow",
};

export default {
  args,
  argTypes: {
    children: {
      description: "Title context",
      control: {
        type: "text",
      },
    },
    colorKey: {
      description: "Back ink color",
      control: { type: "select", options: ["yellow", "green", "blue", "purple", "red"] },
    },
  },
  component: Title,
} as Meta;

export const Default: Story = {};
export const Yellow: Story = {
  args: {
    colorKey: "yellow",
  },
};
export const Green: Story = {
  args: {
    colorKey: "green",
  },
};
export const Blue: Story = {
  args: {
    colorKey: "blue",
  },
};
export const Purple: Story = {
  args: {
    colorKey: "purple",
  },
};
export const Red: Story = {
  args: {
    colorKey: "red",
  },
};
