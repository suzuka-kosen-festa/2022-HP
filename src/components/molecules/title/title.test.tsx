import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./title.stories";

const { Default, Yellow, Blue, Green, Red, Purple } = composeStories(stories);

const options = {
  name: "This is test",
};

describe("(components) molecules/title", () => {
  test("to be Molecules", () => {
    const { container } = render(<Default />);
    expect(container).toBeMolecule();
  });
  test("to be [role=heading]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("heading", options));
  });
  test("(default) take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(yellow) take snap shot", () => {
    const { container } = render(<Yellow />);
    expect(container).toMatchSnapshot();
  });
  test("(blue) take snap shot", () => {
    const { container } = render(<Blue />);
    expect(container).toMatchSnapshot();
  });
  test("(green) take snap shot", () => {
    const { container } = render(<Green />);
    expect(container).toMatchSnapshot();
  });
  test("(red) take snap shot", () => {
    const { container } = render(<Red />);
    expect(container).toMatchSnapshot();
  });
  test("(purple) take snap shot", () => {
    const { container } = render(<Purple />);
    expect(container).toMatchSnapshot();
  });
});
