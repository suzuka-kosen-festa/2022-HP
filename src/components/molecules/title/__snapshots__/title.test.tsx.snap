// Vitest Snapshot v1

exports[`(components) molecules/title > (blue) take snap shot 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <img
      alt="ink"
      class="css-mhf7w5-Title"
      height="580"
      loading="eager"
      src="/statics/Ink3_blue.webp"
      width="592"
    />
    <h1
      class="css-1ueto1"
    >
      This is test
    </h1>
  </div>
</div>
`;

exports[`(components) molecules/title > (default) take snap shot 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <img
      alt="ink"
      class="css-mhf7w5-Title"
      height="580"
      loading="eager"
      src="/statics/Ink3_yellow.webp"
      width="592"
    />
    <h1
      class="css-1ueto1"
    >
      This is test
    </h1>
  </div>
</div>
`;

exports[`(components) molecules/title > (green) take snap shot 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <img
      alt="ink"
      class="css-mhf7w5-Title"
      height="580"
      loading="eager"
      src="/statics/Ink3_green.webp"
      width="592"
    />
    <h1
      class="css-1ueto1"
    >
      This is test
    </h1>
  </div>
</div>
`;

exports[`(components) molecules/title > (purple) take snap shot 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <img
      alt="ink"
      class="css-mhf7w5-Title"
      height="580"
      loading="eager"
      src="/statics/Ink3_purple.webp"
      width="592"
    />
    <h1
      class="css-1ueto1"
    >
      This is test
    </h1>
  </div>
</div>
`;

exports[`(components) molecules/title > (red) take snap shot 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <img
      alt="ink"
      class="css-mhf7w5-Title"
      height="580"
      loading="eager"
      src="/statics/Ink3_red.webp"
      width="592"
    />
    <h1
      class="css-1ueto1"
    >
      This is test
    </h1>
  </div>
</div>
`;

exports[`(components) molecules/title > (yellow) take snap shot 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <img
      alt="ink"
      class="css-mhf7w5-Title"
      height="580"
      loading="eager"
      src="/statics/Ink3_yellow.webp"
      width="592"
    />
    <h1
      class="css-1ueto1"
    >
      This is test
    </h1>
  </div>
</div>
`;
